staging/src/k8s.io/apiserver/pkg/util/flowcontrol/fairqueuing/queueset/types.go

```diff
@@ -104,12 +104,23 @@ func (q *queue) Dequeue() (*request, bool) {
 // GetVirtualFinish returns the expected virtual finish time of the request at
 // index J in the queue with estimated finish time G
 func (q *queue) GetVirtualFinish(J int, G float64) float64 {
-	// The virtual finish time of request number J in the queue
-	// (counting from J=1 for the head) is J * G + (virtual start time).
+	var estimatedServiceTime float64
+	currentIdx := 0
+	q.requests.Walk(func(r *request) bool {
```

Comment:
I was wondering about performance of this method (that is now linear to J).
But looking into codebase, it seems it's only called from a single place:
https://github.com/kubernetes/kubernetes/blob/fddb3adcfd963ef7ee1230bcbb2fd0d324fe8935/staging/src/k8s.io/apiserver/pkg/util/flowcontrol/fairqueuing/queueset/queueset.go#L678
always with J=0.

So to mitigate the potential performance impact if we start calling it for larger J, how about changing the function to sth like:
```
GetVirtualFinishTimeOfFirstRequest(G float64) float64 {
}
```
and leaving the logic generic as you have here but adding a comment that the logic works also for J>0, but before starting to use it for J>0 the performance impact of it has to be carefully evaluated.

Reply:
I am fine with this change, maybe `GetVirtualFinishTimeOfTheOldestRequest` is a more appropriate name? 

Also I would like @MikeSpreitzer to weigh in on why we had `GetVirtualFinish` in the first place, it doesn't look like we will have a scenario where we would need to know the virtual finish time of a request with `J>0`

Reply:
> I am fine with this change, maybe GetVirtualFinishTimeOfTheOldestRequest is a more appropriate name?

I'm always open for better names - that makes sense to me.

Yes - let's wait for Mike.

Reply:
I agree with specializing this function to only get the virtual finish of the oldest request.
`GetVirtualFinishTimeOfTheOldestRequest` would be a fine name.

Reply:
Actually, if we are fully spelling it out, it would be `GetVirtualFinishTimeOfOldestWaitingRequest`.  That's getting a bit too verbose for my taste.  I think `GetNextFinish` would suffice.

Reply:
okay, I picked `GetNextFinish`